tasks/unit-tests/git_api_tests.py

```diff
@@ -0,0 +1,275 @@
+import unittest
+from itertools import cycle
+from unittest import mock
+
+from invoke.exceptions import Exit
+
+from .. import release
+from ..libs.common.github_api import GithubAPI
+from ..libs.common.github_workflows import GithubException, GithubWorkflows
+from ..libs.common.gitlab import Gitlab, get_gitlab_token
+from ..libs.version import Version
+
+##################### MOCKED GITLAB #####################
+
+
+def mocked_502_gitlab_requests(*_args, **_kwargs):
+    class MockResponse:
+        def __init__(self, content, status_code):
+            self.content = content
+            self.status_code = status_code
+
+        def json(self):
+            return self.content
```

Comment:
Maybe you could define `MockResponse` at file scope of the file so that they're less repetitions ! WDYT